### ISSUE TITLE
No way to simulate network outages with FakeServer?

### ISSUE THREAD

**Issue by froodian** [sinonjs/sinon#772]
In reality, when a network outage occurs, both readystatechange and onerror will fire.  Steps to repro:

```
var xhReq = new XMLHttpRequest();
xhReq.addEventListener('readystatechange', function() {
  alert('readystatechange');
});
xhReq.onerror = function () {
  alert('onerror');
};
xhReq.open('POST', '/test', true);
xhReq.send({});
```

Turn off your network connection and run this code - in reality, both handlers will fire.  There doesn't appear to be a way to simulate this case with fakeserver.

**Comment by antoineverger** [comment]
I was looking as well for a solution for that and so far didn´t find any. Did you make any progress on this topic @froodian ?

**Comment by froodian** [comment]
Nope, sorry

**Comment by Standard8** [comment]
In issue #361, there was this potential solution:

    request.respond(0, {}, "");

That doesn't work 100% properly in the currently released code (`onerror` isn't called), but #861 has fixed this in the unreleased code base.
